source/server/listener_manager_impl.cc

```diff
@@ -426,13 +426,23 @@ bool ListenerManagerImpl::addOrUpdateListenerInternal(
   // avoids confusion during updates and allows us to use the same bound address. Note that in
   // the case of port 0 binding, the new listener will implicitly use the same bound port from
   // the existing listener.
-  if ((existing_warming_listener != warming_listeners_.end() &&
-       *(*existing_warming_listener)->address() != *new_listener->address()) ||
-      (existing_active_listener != active_listeners_.end() &&
-       *(*existing_active_listener)->address() != *new_listener->address())) {
-    const std::string message = fmt::format(
-        "error updating listener: '{}' has a different address '{}' from existing listener", name,
-        new_listener->address()->asString());
+  bool active_listener_exists = false;
+  bool warming_listener_exists = false;
+  if (existing_warming_listener != warming_listeners_.end() &&
+      *(*existing_warming_listener)->address() != *new_listener->address()) {
+    warming_listener_exists = true;
```

Comment:
Mind adding a unit test for this case, now they're split out?

Reply:
added a case in the existing test that covers this